pkg/gofr/gofr.go

```diff
@@ -309,87 +311,102 @@ func (a *App) initTracer() {
 
 	traceExporter := a.Config.Get("TRACE_EXPORTER")
 	tracerURL := a.Config.Get("TRACER_URL")
+	authHeader := a.Config.Get("TRACER_AUTH_KEY")
 
-	// deprecated : tracer_host and tracer_port is deprecated and will be removed in upcoming versions
+	// deprecated : tracer_host and tracer_port are deprecated and will be removed in upcoming versions.
 	tracerHost := a.Config.Get("TRACER_HOST")
 	tracerPort := a.Config.GetOrDefault("TRACER_PORT", "9411")
 
-	if tracerURL != "" && traceExporter == "" {
-		a.Logger().Error("missing TRACE_EXPORTER config, should be provided with TRACER_URL to enable tracing")
-		return
+	if tracerURL == "" && (tracerHost != "" || tracerPort != "") {
```

Comment:
I'm unsure why you restored tge default value for the port, but I'm fine with it.

But then, the tracer port is always non empty.

So you would display an error message if tracerURL is empty. So here you can face an error message even if tracer is not used at all.

So please use this

```suggestion
	if tracerURL == "" && tracerHost != "" {
```

Reply:
@srijan-27  think this is a break change  so i revert it